clang/include/clang/Basic/AddressSpaces.h

```diff
@@ -59,6 +59,9 @@ enum class LangAS : unsigned {
   // HLSL specific address spaces.
   hlsl_groupshared,
 
+  // Vulkan specific address spaces.
+  vulkan_private,
+
```

Comment:
Do we need a new address space? This is not an issue specific to these input and output variable. We need to figure out how to handle general static variables. There are some tests that already consider this, but they are DXIL only.

https://github.com/llvm/llvm-project/blob/94eebf721a2f8630412730f51d5071816a686ea0/clang/test/CodeGenHLSL/GlobalDestructors.hlsl#L45

We might need to figure that out first, and than have this PR build on top of that. For DXIL, the static variables are marked as "internal global", and does not use a different address space. Can we do the same? Is the DXIL codegen their real solution or just a temporary implementation?

Reply:
My understanding is we have 2 ways to convey this information from the FE to the BE:
 - using a common IR representation (like `internal global [const]`)
 - using a custom address space.

Right now, the BE considers an "internal global" to have the "Function" storage class, which I believe is wrong. So we could say `internal global` are `Private`.
If for some reasons Intel is against this, we can rely on a custom address space.

The less-invasive change is indeed to use the `internal global`. Which can also be done (Change in SPIRVInstructionSelector.cpp, line 3423 and 3344).
Changing the it doesn't break any tests.

Reply:
It breaks 1 test, but the test seems wrong anyway: test for SPV_INTEL_function_pointers, which is not official IIRC (could only find https://github.com/intel/llvm/blob/sycl/sycl/doc/design/spirv-extensions/SPV_INTEL_function_pointers.asciidoc ) and the spec seems to require the SC to be `CodeSectionINTEL` and the test expects it to be `Function`.
Anyway, if MS approves the overall direction to go through this Ctor/Dtor, I'll make a real PR with those changes, and not use a custom AS